src/store/featureflag.store.ts

```diff
@@ -0,0 +1,64 @@
+import { create } from "zustand";
+import { Dices, LucideIcon } from "lucide-react";
+
+export interface FeatureFlags {
+  gameframe: boolean;
+  moreitems: boolean;
+}
+
+export interface FeatureFlip {
+  id: keyof FeatureFlags;
+  label: string;
+  desc: string;
+  icon: LucideIcon;
+}
+
+const defaultFlags: FeatureFlags = {
+  gameframe: false,
+  moreitems: false,
+};
+
+const featureFlips: FeatureFlip[] = [
+  // UNCOMMENT BELOW!
+  // {
+  //   id: "gameframe",
+  //   label: "Game Frame",
+  //   desc: "This enables external game frame content. NOT IMPLEMENTED - JUST FOR FLAG DEMO.",
+  //   icon: Dices,
+  // },
+  // {
+  //   id: "moreitems",
+  //   label: "More Items",
+  //   desc: "This would be something else!",
+  //   icon: Dices,
+  //},
+];
```

Comment:
What do you think to tie flag with feature like this?
```ts
export enum FeatureFlags {
  GAME_FRAME = 'gameframe',
  MORE_ITEM = 'more_item',
}

export type FeatureFlagsTable = Record<FeatureFlags, boolean>;

const defaultFeatureFlagsTable: FeatureFlagsTable = {
  [FeatureFlags.GAME_FRAME]: false,
  [FeatureFlags.MORE_ITEM]: false,
};

export interface FeatureDescription {
  label: string;
  desc: string;
  icon: string;
}
export type FeatureFlips = Record<FeatureFlags, FeatureDescription>;

const featureFlips: FeatureFlips = {
  [FeatureFlags.GAME_FRAME]: {
    label: 'Game frame',
    desc: 'Game frame desc',
    icon: '...',
  },
  [FeatureFlags.MORE_ITEM]: {
    label: 'MORE_ITEM',
    desc: 'MORE_ITEM',
    icon: '...',
  },
};
```

Reply:
This forces to define all feature flips (type error if not), without the need to re-refine `id`.

Using an enum to be able to add type doc along it, and potentially easier access `FeatureFlags.xxx`